components/layout/display_list/webrender_helpers.rs

```diff
@@ -134,17 +118,18 @@ impl WebRenderDisplayItemConverter for DisplayItem {
                             glyph.advance()
                         };
                         if !slice.glyphs.is_whitespace() {
-                            let glyph_offset = glyph.offset().unwrap_or(Point2D::zero());
-                            let x = (origin.x + glyph_offset.x).to_f32_px();
-                            let y = (origin.y + glyph_offset.y).to_f32_px();
+                            let glyph_offset =
+                                glyph.offset().unwrap_or(Point2D::zero()).to_layout();
```

Comment:
@emilio I am pretty sure that the tests failures on Mac are not intermittent. First I had [the same tests fail](https://github.com/servo/servo/pull/16838#issuecomment-301319272) half a year ago when I tried to change bounds to floating point coordinates. 

The problem is very likely in this code block because it changes the addition from `(Au + Au) as f32` to `f32 + (Au as f32)` and the result happens to be slightly different in some cases and Mac font rendering is affected by this.

The solution is to convert `Text` to `Glyph`es in the display list builder when the precise bounds are still available. (This is needed anyway when servo switches completely to WebRender display lists.)